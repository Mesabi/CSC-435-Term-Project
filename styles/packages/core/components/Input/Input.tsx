import * as React from 'react';
import styled from '@xstyled/styled-components';
import {
  padding,
  PaddingProps,
  borders,
  BorderProps,
  shadow,
  ShadowProps,
} from 'styled-system';

export type InputProps = PaddingProps &
  BorderProps &
  ShadowProps &
  React.HTMLAttributes<HTMLInputElement>;

const Input = styled.input<InputProps>`
  outline: none;
  border: none;
  cursor: text;

  padding: 3 3 5 3;

  color: materialText;
  background-color: inputBackground;

  border-radius: 0;

  border-top-width: 1;
  border-top-style: 1;
  border-top-color: borderDark;

  border-right-width: 0;
  border-bottom-width: 0;

  border-left-width: 1;
  border-left-style: 1;
  border-left-color: borderDark;

  box-shadow: input;

  -webkit-appearance: none;

  ${padding}
  ${borders}
  ${shadow}
`;

export default Input;
