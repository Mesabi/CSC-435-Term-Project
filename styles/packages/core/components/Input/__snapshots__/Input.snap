// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match snapshot 1`] = `
.c0 {
  outline: none;
  border: none;
  cursor: text;
  padding: 3px 3px 5px 3px;
  color: #000000;
  background-color: #ffffff;
  border-radius: 0;
  border-top-width: 1px;
  border-top-style: solid;
  border-top-color: #868a8e;
  border-right-width: 0;
  border-bottom-width: 0;
  border-left-width: 1px;
  border-left-style: solid;
  border-left-color: #868a8e;
  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000, 0.5px 0.5px 0 0.5px #ffffff;
  -webkit-appearance: none;
}

<div>
  <input
    class="c0"
  />
</div>
`;
